ledger-bridge.js

```diff
@@ -34,40 +49,82 @@ export default class LedgerBridge {
         window.parent.postMessage(msg, '*')
     }
 
+    delay (ms) {
+        return new Promise((success) => setTimeout(success, ms))
+    }
+
+    checkTransportLoop (i) {
+        const iterator = i || 0
+        return WebSocketTransport.check(BRIDGE_URL).catch(async () => {
+            await this.delay(TRANSPORT_CHECK_DELAY)
+            if (iterator < TRANSPORT_CHECK_LIMIT) {
+                return this.checkTransportLoop(iterator + 1)
+            } else {
+                throw new Error('Ledger transport check timeout')
+            }
+        })
+    }
+
     async makeApp () {
         try {
-            this.transport = await TransportU2F.create()
-            this.app = new LedgerEth(this.transport)
+            if (this.useLedgerLive) {
+                await WebSocketTransport.check(BRIDGE_URL).catch(async () => {
+                    window.open('ledgerlive://bridge?appName=Ethereum')
+                    await this.checkTransportLoop()
+                    this.transport = await WebSocketTransport.open(BRIDGE_URL)
+                    this.app = new LedgerEth(this.transport)
+                })
+            }
+            else {
+                this.transport = await TransportU2F.create()
+                this.app = new LedgerEth(this.transport)
+            }
         } catch (e) {
             console.log('LEDGER:::CREATE APP ERROR', e)
+            throw e
         }
     }
 
-    cleanUp () {
+    updateLedgerLivePreference (replyAction, useLedgerLive) {
+        this.useLedgerLive = useLedgerLive
```

Comment:
We should probably be calling cleanup here as well, so that the websocket connection doesn't remain open indefinitely when it's not being used.